mltsp/tests/test_predict.py

```diff
@@ -276,14 +357,496 @@ def test_main_predict_tarball_cust_feats():
+                cfg.UPLOAD_FOLDER)
+
+    pred_results_dict = pred.predict(
+        pjoin(cfg.UPLOAD_FOLDER, "215153_215176_218272_218934.tar.gz"),
+        "TEMP_TEST01", "RC", "TEMP_TEST01",
+        metadata_file_path=pjoin(cfg.UPLOAD_FOLDER,
+                                 "215153_215176_218272_218934_metadata.dat"),
+        custom_features_script=None)
+    os.remove(pjoin(cfg.UPLOAD_FOLDER, "215153_215176_218272_218934.tar.gz"))
+    os.remove(pjoin(cfg.UPLOAD_FOLDER,
+                    "215153_215176_218272_218934_metadata.dat"))
+    os.remove(pjoin(cfg.FEATURES_FOLDER, "TEMP_TEST01_features.csv"))
+    os.remove(pjoin(cfg.FEATURES_FOLDER, "TEMP_TEST01_targets.npy"))
+    os.remove(pjoin(cfg.MODELS_FOLDER, "TEMP_TEST01.pkl"))
+    npt.assert_equal(len(pred_results_dict), 4)
+    assert(all(pred_results_dict[fname]["pred_results"] in
+               ['Mira', 'Herbig_AEBE', 'Beta_Lyrae', 'Classical_Cepheid',
+                'W_Ursae_Maj', 'Delta_Scuti', 'RR_Lyrae'] for fname in
+               pred_results_dict))
+
+
+def test_do_model_predictions_ardr():
+    """Test model predictions - ARD Regression"""
+    generate_model_ardr()
+    os.rename(pjoin(cfg.MODELS_FOLDER, "TEMP_TEST01.pkl"),
+              pjoin(cfg.MODELS_FOLDER, "TEST001.pkl"))
+    for suffix in ["targets.npy", "features.csv"]:
+        shutil.copy(
+            pjoin(DATA_PATH, "test_%s" % suffix),
+            pjoin(cfg.FEATURES_FOLDER, "TEST001_%s" % suffix))
+    featset_key = model_key = "TEST001"
+    model_type = "ARDR"
+    features_to_use = ["std_err", "avg_err", "med_err", "n_epochs",
+                       "amplitude"]
+    data_dict = pred.featurize_tsdata(
+        pjoin(DATA_PATH, "dotastro_215153.dat"),
+        None, None, False, features_to_use)
+    pred_results_dict = pred.do_model_predictions(data_dict, model_key,
+                                                  model_type, featset_key,
+                                                  features_to_use, 5)
+    assert("dotastro_215153" in pred_results_dict)
+    assert("std_err" in
+           pred_results_dict["dotastro_215153"]["features_dict"])
+    assert(all(isinstance(pred_results_dict[fname]["pred_results"],
+                          (float, np.float)) for fname in pred_results_dict))
+
+
+def test_main_predict_ardr():
+    """Test main predict function - ARD Regression"""
+
+    generate_model_ardr()
+
+    shutil.copy(pjoin(DATA_PATH, "dotastro_215153.dat"),
+                pjoin(cfg.UPLOAD_FOLDER, "TESTRUN_215153.dat"))
+    shutil.copy(pjoin(DATA_PATH, "TESTRUN_215153_metadata.dat"),
+                cfg.UPLOAD_FOLDER)
+
+    pred_results_dict = pred.predict(
+        pjoin(cfg.UPLOAD_FOLDER, "TESTRUN_215153.dat"),
+        "TEMP_TEST01", "ARDR", "TEMP_TEST01",
+        metadata_file_path=pjoin(cfg.UPLOAD_FOLDER,
+                                 "TESTRUN_215153_metadata.dat"),
+        custom_features_script=None)
+    os.remove(pjoin(cfg.UPLOAD_FOLDER, "TESTRUN_215153.dat"))
```

Comment:
Looks like this kind of thing could be done with a context:

```
with dataset('TESTRUN_215153'):
   ... test code here
```

Then cleanup is also automated.